src/bitcoin-tx.cpp

```diff
@@ -395,27 +395,50 @@ static void MutateTxAddOutMultiSig(CMutableTransaction& tx, const std::string& s
 static void MutateTxAddOutData(CMutableTransaction& tx, const std::string& strInput)
 {
     CAmount value = 0;
+ 
+    // separate [VALUE:]DATA[:ASSET] in string
+    std::vector<std::string> vStrInputParts;
+    boost::split(vStrInputParts, strInput, boost::is_any_of(":"));
 
-    // separate [VALUE:]DATA in string
-    size_t pos = strInput.find(':');
+    // Check that there are enough parameters
+    if (vStrInputParts.empty())
+        throw std::runtime_error("invalid TX output data");
```

Comment:
Again, this can never happen. `boost::split` of an empty string produces a vector with 1 element: the empty string.